hedera-node/src/main/java/com/hedera/services/ledger/interceptors/LinkAwareUniqueTokensCommitInterceptor.java

```diff
@@ -20,26 +20,44 @@
  * ‍
  */
 
-import com.hedera.services.context.SideEffectsTracker;
 import com.hedera.services.ledger.CommitInterceptor;
 import com.hedera.services.ledger.EntityChangeSet;
 import com.hedera.services.ledger.properties.NftProperty;
 import com.hedera.services.state.merkle.MerkleUniqueToken;
+import com.hedera.services.state.submerkle.EntityId;
 import com.hedera.services.store.models.NftId;
 
 /**
  * Placeholder for upcoming work.
  */
-public class UniqueTokensCommitInterceptor implements CommitInterceptor<NftId, MerkleUniqueToken, NftProperty> {
-	public UniqueTokensCommitInterceptor(final SideEffectsTracker sideEffectsTracker) {
-		// No-op
+public class LinkAwareUniqueTokensCommitInterceptor implements CommitInterceptor<NftId, MerkleUniqueToken, NftProperty> {
+	private final UniqueTokensLinkManager uniqueTokensLinkManager;
+
+	public LinkAwareUniqueTokensCommitInterceptor(final UniqueTokensLinkManager uniqueTokensLinkManager) {
+		this.uniqueTokensLinkManager = uniqueTokensLinkManager;
 	}
 
 	/**
 	 * {@inheritDoc}
 	 */
 	@Override
 	public void preview(final EntityChangeSet<NftId, MerkleUniqueToken, NftProperty> pendingChanges) {
-		// No-op
+		final var n = pendingChanges.size();
+		if (n == 0) {
+			return;
+		}
+		for (int i = 0; i < n; i++) {
+			final var entity = pendingChanges.entity(i);
+			final var change = pendingChanges.changes(i);
+			if (entity != null && change != null && change.containsKey(NftProperty.OWNER)) {
```

Comment:
⚠️ What about a `TokenWipe`? Then `change == null`, but `entity.getOwner() ≠ 0.0.0` so we need to update the wiped account's links.

Reply:
TokenWipe I am handling in the wipe transition logic itself